### PR TITLE
Feature/l2 list request status api

### PR DESCRIPTION
**Pull request by simen-rekkedal**
## Related Issue(s)
- #578 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Authentication/Controllers/RequestSystemUserController.cs [hunk 1]
```diff
@@ -227,4 +228,34 @@ public async Task<ActionResult<CreateRequestSystemUserResponse>> ApproveSystemUs
...
     }
+
+    /// <summary>
+    /// Retrieves a list of Status-Response-model for all Requests that the Vendor has for a given system they own.
+    /// </summary>
+    /// <param name="systemId">The system the Vendor wants the list for</param>
+    /// <param name="cancellationToken">The cancellation token</param>
+    /// <returns>Status response model CreateRequestSystemUserResponse</returns>
+    [Authorize(Policy = AuthzConstants.POLICY_SCOPE_SYSTEMREGISTER_WRITE)]
+    [HttpGet("vendor/bysystem/{systemId}")]
```
Comment (acn-dgopa) [inline src/Authentication/Controllers/RequestSystemUserController.cs:239]: I think we need to discuss about the routes, most of our routes are not self explanatory

File: src/Authentication/Controllers/RequestSystemUserController.cs [hunk 2]
```diff
@@ -227,4 +228,34 @@ public async Task<ActionResult<CreateRequestSystemUserResponse>> ApproveSystemUs
...
+    public async Task<ActionResult<CreateRequestSystemUserResponse>> GetAllRequestsForVendor(string systemId, CancellationToken cancellationToken = default)
```
Comment (acn-dgopa) [inline src/Authentication/Controllers/RequestSystemUserController.cs:240]: The model name "CreateRequestSystemResponse" can be changed to something like "RequestSystemResponse" to be more generic

File: src/Persistance/RepositoryImplementations/RequestRepository.cs [hunk 3]
```diff
@@ -207,4 +208,36 @@ private static ValueTask<CreateRequestSystemUserResponse> ConvertFromReaderToReq
...
+
+            command.Parameters.AddWithValue("system_id", systemId);
+
+            return await command.ExecuteEnumerableAsync(cancellationToken)
+                .SelectAwait(ConvertFromReaderToRequest)
+                .ToListAsync(cancellationToken);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Authentication // RequestRepository // GetRequestByInternalId // Exception");
```
Comment (acn-dgopa) [inline src/Persistance/RepositoryImplementations/RequestRepository.cs:239]: method name shoud be updated in comment

File: src/Authentication/Controllers/RequestSystemUserController.cs (hunk 1 above)
Comment (simen-rekkedal) [reply src/Authentication/Controllers/RequestSystemUserController.cs:239]: I will make a note of this, and bring it up with the team.

File: src/Authentication/Controllers/RequestSystemUserController.cs (hunk 2 above)
Comment (simen-rekkedal) [reply src/Authentication/Controllers/RequestSystemUserController.cs]: agreed, changed in update commit

File: src/Persistance/RepositoryImplementations/RequestRepository.cs (hunk 3 above)
Comment (simen-rekkedal) [reply src/Persistance/RepositoryImplementations/RequestRepository.cs]: ok, fixed

**Review by acn-dgopa — APPROVED** [review]

**Comment by acn-dgopa** [comment]
Quality gate still fails, need to add tests?

**Pull request merged**